tests/functional/tests.py

```diff
@@ -122,45 +119,44 @@ def name(self): return self.__name
     @property
     def function(self): return self.__function
     @property
-    def description(self): return self.__description
-    @property
     def args(self): return self.__args
 
     @name.setter
     def name(self, val): self.__name = val
     @function.setter
     def function(self, val): self.__function = val
-    @description.setter
-    def description(self, val): self.__description = val
     @args.setter
     def args(self, val): self.__args = val
 
     def json_report_failure(self, err_msg):
-        # print(err_msg)
-        return json.dumps({'name': self.__name,
+        self.__args = {k: v for k, v in self.__args.items() if v and v != ''}
+        entry = {'name': self.__name,
             'function': self.__function,
-            'description': self.__description,
             'args': self.__args,
-            'duration': round(time.time() - self.__start_time, 2),
+            'duration': int(round((time.time() - self.__start_time)*1000)),
             'alert': self.__alert,
             'message': str(err_msg),
             'error': traceback.format_exc(),
             'status': self.FAIL
-        })
+        }
+        return json.dumps({k: v for k, v in entry.items() if v and v != ''})
 
     def json_report_success(self):
```

Comment:
`json_report_failure()` and ` json_report_success()` are same code expect `err_msg`. You could use lower level function which accepts `err_msg` where `json_report_failure()` and `json_report_success()` uses the function by passing `err_msg` with null or value.